### ISSUE TITLE
Make 'Derived' and 'Actions' optional in the Identify Results Tree panel

### ISSUE THREAD

**Issue by rduivenvoorde** [qgis/QGIS#39385]
When using the i-tool (identify) on several layers, in the 'Top Down' mode, the attribute key/values will be presented in Identify Results panel.
Besides the actual attributes of a features QGIS currently also adds (sometimes A LOT) of so called 'Derived' attributes, like clicked xy, closest xy, parts and parts number etc etc. 

I propose to create a (in profile saved) setting, with which you either show or hide this Derived Attributes.

To enable/disable, there could be:
- add an item in the little Settings menu in the toolbar of the panel (the little tool button)

![Screenshot-20201015102053-365x88](https://user-images.githubusercontent.com/731673/96096422-24e66f00-0ed0-11eb-92cd-f2aa6b88f5e5.png)

- add some kind of little button in the little button bar of the panel (like the tool button)

- there is also an 'Identify'-section in the Settings/Map Tools tab, but I do not think this should be hidden for users, it should be easy to enable/disable it as users also find this information helpful
(see https://lists.osgeo.org/pipermail/qgis-developer/2020-October/062391.html)

- we could make 2 options, one to hide the derived ones and one to hide the actions, but to start (and to me) it is ok to do this in one go. 
Another option would be to make these attributes not so major, as in put them on bottom of the list, or grey them out.

Below is a quick test, to see what happens if I disable the derived attributes.
These are two vector layers, a point and a polygon (left is with and right is without derived info):

![Screenshot-20201014172526-868x856](https://user-images.githubusercontent.com/731673/96096079-b86b7000-0ecf-11eb-8693-45d9b8faa153.png)

Note that there are 4 places in code (for vector, raster, mesh and vector tile layers) to fix this.

Below a screenie of a identify panel with all four types of layers in it:

![Screenshot-20201015101331-453x1216](https://user-images.githubusercontent.com/731673/96097301-13519700-0ed1-11eb-97ed-2427de003723.png)

**Comment by jmckenna** [comment]
Personally I constantly check my settings in /Settings/Options/ and I honestly had no idea that the identify panel has its own settings that can only be accessed with the little tool icon at the top of the identify panel (I have never noticed that icon before).   

I'd prefer that this setting be available both in /Settings/Options/Map Tools/, as well as directly in the panel settings.

**Comment by jmckenna** [comment]
I personally find the little tool icon in the identify panel very "hidden" to users, and most users will be used to checking the main settings in /Settings/Options.        

Thoughts?

**Comment by jmckenna** [comment]
Actually, maybe it is the icon itself (the wrench/tool icon in my brain means something like 'query builder' or 'build' something, not settings.  Settings icon in my brain (and most software/smart phones) is a cog wheel.

Hope my feedback helps, even if no one agrees with me, it at least gives you my feedback.  

Finally, thanks for this enhancement and your efforts.  Really appreciate it.

**Comment by rduivenvoorde** [comment]
@jmckenna well about the 'wrench/tool' icon. That is another 'papercut' GUI thing that I would want to raise. In different parts of QGIS, the 'settings' have different icons: a wrench, or this gear icon: 
Left is the Temporal Controller with a gear icon, right is the Results panel with a wrench, both pointing to some kind of 'Settings':

![Screenshot-20201015144806-997x143](https://user-images.githubusercontent.com/731673/96128303-7523f800-0ef5-11eb-9244-4f3f24559f21.png)

About where to actually do the setting. I'm OK to do it in both places.

**Comment by jmckenna** [comment]
Agreed to use the same icon for settings throughout QGIS, as the temporal controller/cog/gear.  Sorry for bringing it up in this ticket, but in my mind it is related to this discussion.

**Comment by andreasneumann** [comment]
And to make things a bit worse - the cog-wheel is also used for
- the processing toolbox: 
![image](https://user-images.githubusercontent.com/884476/96129392-dc8e7780-0ef6-11eb-9bd8-ea4e19376a40.png)
- triggering an action: 
![image](https://user-images.githubusercontent.com/884476/96129522-034cae00-0ef7-11eb-8200-c383f343535c.png)

Seems like the cog is some sort of "placeholder" icon where the dev didn't find anything better or more meaningful ...

**Comment by rduivenvoorde** [comment]
I finished a PR  #39399 to add a bool setting behind the little tool button. Sorry @jmckenna but adding it in the main settings is a little over my head, and in my opinion this boolean should be there where you are working with the panel, so I decided to keep it like this.
About the cog/wrench/tool buttons, thanks for the input, but let's make that another issue, because that needs some discussion (I think) about what are 'options', 'settings' etc, and which icons should be used where etc etc... I'm really in favour of such discussions but let's not glue it to this issue.
